tests/unit/specification/parser_test.py

```diff
@@ -2208,6 +2209,85 @@ def test_create_model_checksum() -> None:
     }
 
 
+@pytest.mark.parametrize(
+    "spec,byte_order",
+    [
+        (
+            """
+                package Test is
+                   type T is mod 2**8;
+                   type M is
+                      message
+                         F1 : T;
+                         F2 : T;
+                         F3 : T;
+                     end message
+                        with Byte_Order => Low_Order_First;
+                 end Test;
+            """,
+            model.ByteOrder.LOW_ORDER_FIRST,
+        ),
+        (
+            """
+                package Test is
+                   type T is mod 2**8;
+                   type M is
+                      message
+                         F1 : T;
+                         F2 : T;
+                         F3 : T;
+                     end message
+                        with Byte_Order => High_Order_First;
+                 end Test;
+            """,
+            model.ByteOrder.HIGH_ORDER_FIRST,
+        ),
+        (
+            """
+                package Test is
+                   type T is mod 2**8;
+                   type M is
+                      message
+                         F1 : T;
+                         F2 : T;
+                         F3 : T;
+                     end message;
+                 end Test;
+            """,
+            model.ByteOrder.HIGH_ORDER_FIRST,
+        ),
+        (
+            """
+                package Test is
+                    type T is mod 2**8;
+                    type M is
+                        message
+                            F1 : T;
+                            F2 : T;
+                            C1 : T;
+                            F3 : T;
+                            C2 : T
+                            then F4
+                            if C1'Valid_Checksum and C2'Valid_Checksum;
+                        F4 : T;
+                    end message
+                        with Checksum => (C1 => (F3, F1'First .. F2'Last),
+                                        C2 => (C1'Last + 1 .. C2'First - 1)),
+                             Byte_Order => Low_Order_First;
```

Comment:
Wrong indentation.

Reply:
done

Reply:
The `then F4 if ...` is still misaligned.

Reply:
in fact all the test inputs were misaligned ... hopefully fixed

Reply:
LGTM.